baseclasses/chemical_energy/uvvisconcentrationdetection.py

```diff
@@ -0,0 +1,106 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+
+from nomad.metainfo import Quantity, Reference, Section, SubSection, Datetime
+from nomad.datamodel.metainfo.basesections import Analysis, SectionReference
+
+from nomad.datamodel.metainfo.plot import PlotSection, PlotlyFigure
+import plotly.graph_objects as go
+
+from baseclasses.solar_energy import UVvisMeasurement
+
+class UVvisReference(SectionReference):
+    reference = Quantity(
+        type=Reference(UVvisMeasurement.m_def),
+        a_eln=dict(component='ReferenceEditQuantity', label='UVvis Measurement'))
+
+
+class UVvisConcentrationDetection(Analysis, PlotSection):
+    m_def = Section(label_quantity='name')
+
+    inputs = Analysis.inputs.m_copy()
+    inputs.section_def = UVvisReference
+
+    material_name = Quantity(
+        type=str,
+        a_eln=dict(component='StringEditQuantity'))
+
+    minimum_peak_value = Quantity(
+        type=np.dtype(np.float64),
+        description='The calibration curve can only be applied to peak intensities that are higher than this value.')
+
+    maximum_peak_value = Quantity(
+        type=np.dtype(np.float64),
+        description='The calibration curve can only be applied to peak intensities that are lower than this value.')
+
+    slope = Quantity(
+        type=np.dtype(np.float64),
+        description='The slope of the calibration curve.',
+        a_eln=dict(component='NumberEditQuantity'))
+
+    intercept = Quantity(
+        type=np.dtype(np.float64),
+        description='The intercept of the calibration curve.',
+        a_eln=dict(component='NumberEditQuantity'))
+
+    r2 = Quantity(
+        type=np.dtype(np.float64),
+        description='The R**2 to assess how well the calibration curve reflects the underlying intensity-concentration values.')
+
+    def normalize(self, archive, logger):
+        super(UVvisConcentrationDetection, self).normalize(archive, logger)
+
+        peak_values = []
+        concentrations = []
+
+        for uvvis_reference in self.inputs:
+            for uvvisdata in uvvis_reference.reference.measurements:
+                if uvvisdata.concentration is None or uvvisdata.peak_value is None:
+                    logger.error('Please provide concentration and area data for each UVvis Measurement.')
+                else:
+                    concentrations.append(uvvisdata.concentration)
+                    peak_values.append(uvvisdata.peak_value)
+
+        if len(peak_values) > 0:
+            self.minimum_peak_value = min(peak_values)
+            self.maximum_peak_value = max(peak_values)
+
+            try:
+                concentration_values = np.array([measure.magnitude for measure in concentrations])
```

Comment:
there is also this: 
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.linregress.html might be easier?

Reply:
Somehow I only looked for the optinons in numpy. Changed it now. Thanks :)